06_puppy/brvishnu92/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
+	id    uint32
+	breed string
+	color string
+	value string
+}
+
+type Storer interface {
+	CreatePuppy(*Puppy)
+	ReadPuppy(id uint32) *Puppy
+	UpdatePuppy(puppy *Puppy)
+	DeletePuppy(id uint32) bool
+	HasPuppy(id uint32) bool
+}
+
+type MapStore map[uint32]Puppy
+
+type SynStore struct {
+	store *sync.Map
+}
+
+func (ms MapStore) CreatePuppy(pup *Puppy) {
+	if !ms.HasPuppy(pup.id) {
+		ms[pup.id] = *pup
+	}
+}
+func (ms MapStore) ReadPuppy(id uint32) *Puppy {
+	if puppy, has := ms[id]; has {
```

Comment:
use `ok` instead of `has` ([comma go idiom](https://golang.org/doc/effective_go.html#maps))